Documentation/Diagnostics/PH2129.md

```diff
@@ -0,0 +1,55 @@
+# PH2129: Return immutable collections
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2129 |
+| Category  | [Maintainability](../Maintainability.md) |
+| Analyzer | [ReturnImmutableCollectionsAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/ReturnImmutableCollectionsAnalyzer.cs)
+| CodeFix  | Yes |
+| Severity | Error |
+| Enabled By Default | Yes |
+
+## Introduction
+
+Returned objects from public methods should be protected against mutation from outside your class. Therefore, return only immutable or readonly collections from a public method, otherwise these collections can be changed by the caller without your code noticing.
+
+The list of mutable collections that are checked are:
+* `System.Collections.Generic.List<>`
+* `System.Collections.Generic.Queue<>`
+* `System.Collections.Generic.SortedList<>`
+* `System.Collections.Generic.Stack<>`
+* `System.Collections.Generic.Dictionary<>`
+* `System.Collections.Generic.IList<>`
+* `System.Collections.Generic.IDictionary<>`
+
+## How to solve
+
+Returned collection types from public methods should be either:
+1. `IEnumerable`
+2. one of the `IReadOnly` interfaces for these collections
```

Comment:
IReadOnly's contents are editable by the caller... Maybe explain the different here and/or point to MS documentation.